jcl/src/openj9.dtfj/share/classes/com/ibm/dtfj/javacore/parser/j9/section/environment/EnvironmentSectionParser.java

```diff
@@ -254,24 +257,43 @@ private void parseUserArgs() throws ParserException {
 					try {
 						fRuntimeBuilder.addVMInitArgs();
 					} catch (BuilderFailureException e) {
-						handleError("Failed to add JavaVMInitArgs to builder: ", e);
+						handleError("Failed to add VM init args to builder: ", e);
 					}
 				}
 				String argString = results.getTokenValue(ARG_STRING);
 				long extraInfo = results.getLongValue(ARG_EXTRA);
 				try {
-					if (extraInfo == IBuilderData.NOT_AVAILABLE) {
-						if (argString != null) {
-							fRuntimeBuilder.addVMOption(argString);
-						}
-					} else {
-						fRuntimeBuilder.addVMOption(argString, extraInfo);
+					if (argString != null) {
+						fRuntimeBuilder.addVMInitOption(argString, extraInfo);
 					}
 				} catch (BuilderFailureException e) {
-					handleError("Failed to add VM option to builder: " + argString + " ", e);
+					handleError("Failed to add VM init option to builder: " + argString + " ", e);
 				}
 			}
 		}
+
+/*[IF CRIU_SUPPORT]*/
+		if (null != (results = processTagLineOptional(T_1CIRESTARGS))) {
```

Comment:
Please separate the assignment from the test.

Reply:
Moved assignment out of the if statement.